### PR TITLE
tests: Add AccessMask2 tests

### PR DESCRIPTION
**Pull request by sjfricke**
About to make PR for something related to sync2 and pulled these extra tests out into own PR

### CODE REVIEW

File: tests/vklayertests_buffer_image_memory_sampler.cpp [hunk 1]
```diff
@@ -15499,6 +15499,248 @@ TEST_F(VkLayerTest, TestInvalidBarrierQueues) {
...
+
+    SetTargetApiVersion(VK_API_VERSION_1_1);
+    AddRequiredExtensions(VK_KHR_SYNCHRONIZATION_2_EXTENSION_NAME);
+    ASSERT_NO_FATAL_FAILURE(InitFramework());
+    if (!AreRequiredExtensionsEnabled()) {
+        GTEST_SKIP() << RequiredExtensionsNotSupported() << " not supported";
+    }
+    if (!CheckSynchronization2SupportAndInitState(this)) {
+        printf("%s Synchronization2 not supported, skipping test\n", kSkipPrefix);
+        return;
```
Comment (ncesario-lunarg) [inline tests/vklayertests_buffer_image_memory_sampler.cpp:15513]: `GTEST_SKIP`?

File: tests/vklayertests_buffer_image_memory_sampler.cpp [hunk 2]
```diff
@@ -15499,6 +15499,248 @@ TEST_F(VkLayerTest, TestInvalidBarrierQueues) {
...
+    ASSERT_NO_FATAL_FAILURE(InitFramework());
+    if (DeviceValidationVersion() < VK_API_VERSION_1_1) {
+        GTEST_SKIP() << "Test requires at least Vulkan 1.1.";
+    }
+    if (!AreRequiredExtensionsEnabled()) {
+        GTEST_SKIP() << RequiredExtensionsNotSupported() << " not supported";
+    }
+    if (!CheckSynchronization2SupportAndInitState(this)) {
+        printf("%s Synchronization2 not supported, skipping test\n", kSkipPrefix);
+        return;
```
Comment (ncesario-lunarg) [inline tests/vklayertests_buffer_image_memory_sampler.cpp:15706]: `GTEST_SKIP`?

**Review by ncesario-lunarg — COMMENTED** [review]
Just the 2 `return`s -> `GTEST_SKIP`, but otherwise LGTM!

File: tests/vklayertests_buffer_image_memory_sampler.cpp (hunk 2 above)
Comment (sjfricke) [reply tests/vklayertests_buffer_image_memory_sampler.cpp]: did one better and just made commit to update all the spots using this (which I probably copied/pasted from)

**Review by ncesario-lunarg — APPROVED** [review]

**Pull request merged**